pkg/jsonnet/find_importers.go

```diff
@@ -220,8 +220,24 @@ func findImporters(root string, searchForFile string, chain map[string]struct{})
 		}
 	}
 
-	importersCache[key] = importers
-	return importers, nil
+	// If we've found a vendored file, check that it's not overridden by a vendored file in the environment root
+	// In that case, we only want to keep the environment vendored file
+	var filteredImporters []string
+	rootVendor := filepath.Join(root, "vendor")
+	if strings.HasPrefix(searchForFile, rootVendor) {
+		for _, importer := range importers {
+			relativePath := strings.TrimPrefix(searchForFile, rootVendor)
```

Comment:
could be `filepath.Rel`, that would deal with the trailing slash on the below line too

(which is good, always better to not hardcode the separator)